From 1e0b46a476c1c950e22403a2d8950f5c97a1a0fe Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Fri, 5 Jun 2020 11:35:25 -0700
Subject: [PATCH] CMakeLists.txt: prevent building zstd with ExternalProject

* use pkg-config, because zstd doesn't provide cmake find

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>

Rebase patch on rolling release.

Signed-off-by: Sandeep Gundlupet Raju <sandeep.gundlupet-raju@amd.com>
---
 CMakeLists.txt                      | 54 +++--------------------------
 cmake_minimum_required_2.8.12.patch | 26 --------------
 no_internal_headers.patch           | 27 ---------------
 3 files changed, 4 insertions(+), 103 deletions(-)
 delete mode 100644 cmake_minimum_required_2.8.12.patch
 delete mode 100644 no_internal_headers.patch

Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -8,19 +8,14 @@ list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE
 # We need at least VERSION 1.5.5, version check is done in Findzstd.cmake
 # The platform with the oldest supported version is RHEL-9,
 # which has zstd 1.5.1.  Make sure we have at least that version.
-find_package(zstd 1.5.1 QUIET)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(ZSTD REQUIRED libzstd>=1.5.5)
 
-ament_vendor(zstd_vendor
-  SATISFIED ${zstd_FOUND}
-  VCS_URL https://github.com/facebook/zstd.git
-  VCS_VERSION v1.5.5
-  SOURCE_SUBDIR build/cmake
-  CMAKE_ARGS
-    -DZSTD_BUILD_STATIC:BOOL=OFF
-    -DZSTD_BUILD_SHARED:BOOL=ON
-    -DZSTD_BUILD_PROGRAMS:BOOL=OFF
-    PATCHES patches
-)
+if(NOT ZSTD_FOUND)
+  message(STATUS "Zstd not found, missing dependency or version less than 1.5.5, found ${ZSTD_VERSION}")
+else()
+  message(STATUS "Found Zstd, skipping build.")
+endif()
 
 install(DIRECTORY cmake DESTINATION share/${PROJECT_NAME})
 
